config-overrides.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+const webpack = require('webpack');
+
+module.exports = function override(config, env) {
+  if (env === 'production') {
+    config.plugins.push(
+      // prevent code-splitting to allow the URL path-prefix for the site to be
+      // configurable at runtime
+      // https://github.com/jaegertracing/jaeger-ui/issues/42
+      new webpack.optimize.LimitChunkCountPlugin({
```

Comment:
so which way are we going, global search&replace of the path in the query service or using this plugin to not split the js files?

Reply:
@yurishkuro Still up for debate.

The main considerations:

- **Primarily** - The exact form the loading of JS chunks takes (which is the site of the search/replace) has an unknown level of risk of changing
- The search/replace in the JS files is not as "clean" as the others
- The main.js filesize is ~2M, on disk (not sure to what extent that is much of a consideration)

Note: every *.js file, not just main.NNNN.js, must be checked for the `static/js` path and prefixed if found

So, my main concern is around maintainability. 

The above inconveniences allow us to avert a likely hit on initial page-load time, possibly a ~5% increase.

The ideal scenario, IMO, is to do the global replace and if maintaining it is a PITA then revert and go with the plugin approach to not split the files. Not sure how feasible that approach is on the query-service side.

Would that work for you?

Reply:
Yes, I would suggest trying global replace first.

Reply:
Sounds good. This PR needs some rework, then.